src/compiler/mixin.cc

```diff
@@ -56,28 +83,53 @@ static void add_stubs(ir::Class* klass) {
       }
       auto original_parameters = method->parameters();
       ASSERT(original_parameters.length() == arity);
-      auto stub_parameters = ListBuilder<ir::Parameter*>::allocate(arity);
-      auto forward_arguments = ListBuilder<ir::Expression*>::allocate(arity);
-      for (int i = 0; i < arity; i++) {
-        auto original_parameter = original_parameters[i];
-        ASSERT(original_parameter->index() == i);
-        auto stub_parameter = _new ir::Parameter(original_parameter->name(),
-                                                 original_parameter->type(),
-                                                 original_parameter->is_block(),
-                                                 original_parameter->index(),
-                                                 original_parameter->has_default_value(),
-                                                 original_parameter->range());
-        stub_parameters[i] = stub_parameter;
-        forward_arguments[i] = _new ir::ReferenceLocal(stub_parameter, 0, range);
-      }
-
-      auto forward_call = _new ir::CallStatic(_new ir::ReferenceMethod(method, range),
-                                              shape.to_equivalent_call_shape(),
-                                              forward_arguments,
-                                              range);
-
+      auto stub_parameters = duplicate_parameters(original_parameters);
       ir::MethodInstance* stub;
-      if (method->is_IsInterfaceOrMixinStub()) {
+
+      if (method->is_FieldStub()) {
+        // Copy of what's happening in `resolver_method`.
```

Comment:
It isn't very clear why the code cannot be shared better. Maybe refactor this or improve the comment to explain why this is a copy?

Reply:
It's not completely clear-cut.
It would be possible to share the code, but I don't really want to move the code out of the resolver_method file. Almost all of the IR generation is happening there.
On the other hand I don't really want to call from here into the resolver_method either.
Also, they are just slightly different (like where the `this` comes from, ...).
Finally, the resolver_method has some convenience functions that don't exist here.

I tried once to create a shared file and decided that it wasn't worth it. But I could be convinced otherwise.